pontoon/base/tests/test_views.py

```diff
@@ -123,6 +123,34 @@ def test_not_authed_public_project(self):
         # I'd assertTemplateUsed here but it doesn't work on non-DTL
         # templates.
 
+    def test_locale_agnostic_user_authed(self):
+        """
+        User is authenticated and Userprofile.custom_homepage defined,
+        redirect to Userprofile.custom_homepage
+        """
+        ProjectFactory.create(slug='valid-project')
+        LocaleFactory.create(code='custom-locale')
+        response = self.client.get('custom-locale/valid-project/path')
+        assert_equal(response.status_code, 200)
```

Comment:
You need to test that the redirect works as expected, not that the translate view status code is 200:

```python
response = self.client.get('/projects/valid-project/fake/path.po/')
self.assertRedirects(response, '/custom-locale/valid-project/fake/path.po/')
```

To make the test pass, you'll need to [mock](http://fgimian.github.io/blog/2014/04/10/using-the-python-mock-library-to-fake-regular-functions-during-tests/) `user.is_authenticated()` and `user.profile.custom_homepage` before that.